src/Examples/Flowsheets/power_generation/rsofc/rsofc_soec_example.ipynb

```diff
@@ -0,0 +1,2131 @@
+    "<img src='rsofc_soec_mode_PFD.svg' width=\\\"300\\\" height=\\\"300\\\">"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7e89315f",
+   "metadata": {},
+   "source": [
+    "## 1. Introduction\n",
+    "The figure above shows the process flow diagram of the rSOC plant in SOEC mode. The plant consists of five major sections: the ASU, the SOEC power island, the CPU, the HRSG and steam cycle.\n",
+    "\n",
+    "In the SOEC mode, the rSOC operates by using steam as fuel to produce Hydrogen on its fuel side (cathode side), while air is used as sweep gas to remove diffused oxygen from its air side (anode side). To provide air to the air side of the rSOC, ambient air is compressed in an air blower and preheated via two preheaters to reach the required temperature to sweep through the air side. The oxygen rich air stream which exits the air side of the rSOC is used to preheat the fresh air feed in the first air preheater, subsequently it is split in a splitter and used to preheat oxygen from the ASU in the oxygen preheater and natural gas feed in the natural gas preheater. Finally, the oxygen rich air streams from both the oxygen and natural gas preheaters are sent to the HRSG where they are further cooled before release to the atmosphere.\n",
+    "\n",
+    "This flowsheet example provides an off-design model for the rSOC in SOEC mode for hydrogen production up to **5 kg/s** with 98% CO2 capture."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "be55b226",
+   "metadata": {},
+   "source": [
+    "## 2. Import Required Modules\n",
+    "- **pyomo environment**. The pyomo environment\n",
+    "- **rsofc_soec_flowsheet**. Contains functions for the work flow of building and solving the flowsheet. See the rsofc_soec_flowsheet.py file for more details.\n",
+    "- **rsofc_costing**. Contains costing methods for the soec and sofc operating modes of the reversible sofc. See the rsofc_costing.py file for more details.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "1b17226a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pyomo.environ as pyo\n",
+    "import rsofc_soec_flowsheet as rsoec\n",
+    "import rsofc_costing as rsofc_cost"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6b39b2fe",
+   "metadata": {},
+   "source": [
+    "## 3. Build and initialize the flowsheet\n",
+    "- A pyomo concrete model is built\n",
+    "- The **get_model** function in the **rsofc_soec_flowsheet.py** file takes the concrete model as an argument then sets up a flowsheet called **m.soec_fs**, and all the required unit models to it. Finally, the completed flowsheet is initialized. The concrete model and solver options are then returned."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "ceb3d54e",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "WARNING (W1002): Setting Var\n",
```

Comment:
These Pyomo warnings should be addressed - it is never good practice to set a mole fraction to exactly zero in any case. You should use a sufficiently small number instead.

There are also a large number of warnings about missing scaling factors which clutter the output, and don;t really look good to the user.